src/mlpack/tests/decision_tree_regressor_test.cpp

```diff
@@ -0,0 +1,1047 @@
+//   for (size_t i = dataset.n_cols; i < dataset.n_cols + 1000; ++i)
+//     weights[i] = math::Random(0.0, 0.01); // Low weights for false points.
+
+//   // Now build the decision tree.  I think the syntax is right here.
+//   DecisionTreeRegressor<MADGain> d(data, fullLabels, weights);
+
+//   // Now we can check that we get good performance on the VC2 test set.
+//   arma::mat testData;
+//   arma::Row<double> testLabels;
+//   if (!data::Load("vc2_test.csv", testData))
+//     FAIL("Cannot load test dataset vc2_test.csv!");
+//   if (!data::Load("vc2_test_labels.txt", testLabels))
+//     FAIL("Cannot load labels for vc2_test_labels.txt!");
+
+//   arma::Row<double> predictions;
+//   d.Predict(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+
+//   // Figure out the accuracy.
+//   double accuracy = R2Score(predictions, testLabels);
+
+//   REQUIRE(accuracy > 0.75);
+// }
+
+// /**
+//  * Test that we can build a decision tree using information gain on a simple
+//  * categorical dataset using weights, with low-weight noise added.
+//  */
+// TEST_CASE("CategoricalInformationGainWeightedBuildTest", "[DecisionTreeTest]")
+// {
+//   arma::mat d;
+//   arma::Row<size_t> l;
+//   data::DatasetInfo di;
+//   MockCategoricalData(d, l, di);
+
+//   // Split into a training set and a test set.
+//   arma::mat trainingData = d.cols(0, 1999);
+//   arma::mat testData = d.cols(2000, 3999);
+//   arma::Row<size_t> trainingLabels = l.subvec(0, 1999);
+//   arma::Row<size_t> testLabels = l.subvec(2000, 3999);
+
+//   // Now create random points.
+//   arma::mat randomNoise(4, 2000);
+//   arma::Row<size_t> randomLabels(2000);
+//   for (size_t i = 0; i < 2000; ++i)
+//   {
+//     randomNoise(0, i) = math::Random();
+//     randomNoise(1, i) = math::Random();
+//     randomNoise(2, i) = math::RandInt(4);
+//     randomNoise(3, i) = math::RandInt(2);
+//     randomLabels[i] = math::RandInt(5);
+//   }
+
+//   // Generate weights.
+//   arma::rowvec weights(4000);
+//   for (size_t i = 0; i < 2000; ++i)
+//     weights[i] = math::Random(0.9, 1.0);
+//   for (size_t i = 2000; i < 4000; ++i)
+//     weights[i] = math::Random(0.0, 0.001);
+
+//   arma::mat fullData = arma::join_rows(trainingData, randomNoise);
+//   arma::Row<size_t> fullLabels = arma::join_rows(trainingLabels, randomLabels);
+
+//   // Build the tree.
+//   DecisionTree<InformationGain> tree(fullData, di, fullLabels, 5, weights, 10);
+
+//   // Now evaluate the accuracy of the tree.
+//   arma::Row<size_t> predictions;
+//   tree.Classify(testData, predictions);
+
+//   REQUIRE(predictions.n_elem == testData.n_cols);
+//   size_t correct = 0;
+//   for (size_t i = 0; i < testData.n_cols; ++i)
+//     if (testLabels[i] == predictions[i])
+//       ++correct;
+
+//   // Make sure we got at least 70% accuracy.
+//   const double correctPct = double(correct) / double(testData.n_cols);
+//   REQUIRE(correctPct > 0.70);
+// }
```

Comment:
Should we remove these two tests?

Reply:
I think we can adapt these tests to use MADGain. All the tests currently use MSEGain. I think there should be at least one or two tests where the tree is trained on MADGain. What do you think?

Reply:
I agree, we should test both gain functions for sure.